### ISSUE TITLE
Array type does not work

### ISSUE THREAD

**Issue by lucas2xl** [elysiajs/elysia-swagger#39]
When an array is passed in the response it ends up not recognizing it

`response: t.Array(userDTO)`

<img width="1419" alt="Screenshot 2023-09-28 at 10 11 55" src="https://github.com/elysiajs/elysia-swagger/assets/65921350/89c9f69c-c09f-4712-b65e-3ebb673736a6">

Now if an object is passed to be surrounded by an array, it works perfectly.

`response: userDTO`

<img width="1419" alt="Screenshot 2023-09-28 at 10 12 07" src="https://github.com/elysiajs/elysia-swagger/assets/65921350/501b8df5-7e3d-431b-92bd-eebbee418788">

**Comment by kareemmahlees** [comment]
Would like to work on this, assign it to me please 🙂

**Comment by mel-mouk** [comment]
Hi,
I found a temporary solution on the Discord. Thought I'd share it here in case someone else ends up here first like me.

If you use it directly like this it doesn't work :
```const routes = new Elysia()
  .get('/', async () => { }, {
       response: t.Array(t.Object({ something: t.String() })
  })```
  
  But if you do : 
  ```
  const routes = new Elysia()
  .model({ someDto: t.Array(t.Object({ something: t.String() }))
  .get('/', async () => { }, {
       response: 'someDto'
  })
  ```
  
  then it works

**Comment by mpandzo** [comment]
Thanks @mel-mouk !

For someone looking to control the response description as well as the other parts of the schema (like marking query parameters as not required), here is an example of my approach:

```
import { Elysia, t } from 'elysia';

const ItemType = {
  name: t.String(),
};

const ItemResponseType = t.Object(ItemType , { title: 'Item' });
ItemResponseType.description = 'Retrieve a single item';

const ItemsResponseType = t.Array(
  t.Object(ItemType, {
    title: 'Item',
  }), {
  title: 'Items',
});
ItemsResponseType.description = 'Retrieve list of items';

const ItemsQuerySchema = t.Object({
  name: t.String(),
});
ItemsQuerySchema.required = [];

const sampleItem = {
  name: 'My item',
};

export const routes = new Elysia()
  .model({
    itemResponseType: ItemResponseType,
    itemsResponseType: ItemsResponseType
  })
  .get('/items', () => [sampleItem], {
    type: 'application/json',
    response: 'itemsResponseType',
    query: ItemsQuerySchema,
  })
  .get('/items/:id', () => sampleItem, {
    type: 'application/json',
    response: 'itemResponseType'
  })
```

**Comment by alzalabany** [comment]
so.. is this a bug or expected that array only works with model ?

**Comment by SaltyAom** [comment]
Fixed in d3aecbc, published on 0.8.5

<img width="1920" alt="Screenshot 2567-01-24 at 17 10 05" src="https://github.com/elysiajs/elysia-swagger/assets/35027979/79958079-b1b7-4161-afbd-8d1164ee5343">


![image](https://github.com/elysiajs/elysia-swagger/assets/35027979/e291e6d6-675b-40df-bf55-4edac0c70209)
![image](https://github.com/elysiajs/elysia-swagger/assets/35027979/12cd7a7e-eae5-42b4-b47a-80269e3bbc20)

Feels free to reopen if the issue isn't resolved after updating to 0.8.5, closing.
